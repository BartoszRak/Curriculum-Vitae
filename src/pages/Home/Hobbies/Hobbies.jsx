import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { withStyles, Typography } from '@material-ui/core'

import assasinBugUrl from '~assets/images/assasin-bug.jpg'
import codingUrl from '~assets/images/coding.jpg'

import style from './Hobbies.style'

export function Hobbies({ classes }) {
  return (
    <React.Fragment>
      <div className={classes.root} id="hobbies">
        <Typography className={classes.title} color="secondary" variant="h1">
          <FormattedMessage id="pages.home.hobbies.title" />
          <span aria-hidden className={classes.emoji}>ðŸ™„</span>
        </Typography>
        <div className={classes.tilesHolder}>
          <div className={classes.tile}>
            <Typography variant="h2">
              <FormattedMessage id="pages.home.hobbies.tiles.insects.title" />
            </Typography>
            <Typography className={classes.tileDescription} variant="body1">
              <FormattedMessage id="pages.home.hobbies.tiles.insects.description" />
            </Typography>
            <img alt="Assasing bug" className={classes.tileImage} src={assasinBugUrl} />
          </div>
          <div className={classes.tile}>
            <Typography variant="h2">
              <FormattedMessage id="pages.home.hobbies.tiles.programming.title" />
            </Typography>
            <Typography className={classes.tileDescription} variant="body1">
              <FormattedMessage id="pages.home.hobbies.tiles.programming.description" />
            </Typography>
            <img alt="Coding" className={classes.tileImage} src={codingUrl} />
          </div>
        </div>
      </div>
    </React.Fragment>
  )
}

Hobbies.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(Hobbies)
