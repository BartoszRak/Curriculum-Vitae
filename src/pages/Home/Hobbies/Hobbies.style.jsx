export default ({ breakpoints, palette, spacing }) => ({
  emoji: {
    fontSize: '2rem',
    [breakpoints.down('md')]: {
      fontSize: '1.5rem',
    },
  },
  root: {
    width: '100%',
  },
  tile: {
    alignItems: 'center',
    background: palette.secondary.main,
    borderRadius: spacing.unit * 0.5,
    breakInside: 'avoid-column',
    columnBreakInside: 'avoid',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    minHeight: 200,
    padding: spacing.unit * 3,
    pageBreakInside: 'avoid',
    width: '100%',
  },
  tileDescription: {
    marginTop: spacing.unit * 1.5,
    width: '100%',
  },
  tileImage: {
    borderRadius: spacing.unit * 0.5,
    marginTop: spacing.unit * 1.5,
    objectFit: 'cover',
    position: 'relative',
    width: '100%',
  },
  tilesHolder: {
    columnGap: 80,
    [breakpoints.up('md')]: {
      columnCount: 2,
    },
    [breakpoints.down('md')]: {
      columnCount: 1,
    },
  },
  title: {
    marginBottom: spacing.unit * 4,
    [breakpoints.down('md')]: {
      fontSize: '2rem',
    },
  },
})
