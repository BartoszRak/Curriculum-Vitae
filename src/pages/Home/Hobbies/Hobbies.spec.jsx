import React from 'react'
import { shallow } from 'enzyme'

import { Hobbies } from './Hobbies'

const initComponent = overrides => {
  const mockProps = {
    classes: {},
  }
  const mockMethods = {}
  const wrapper = shallow(<Hobbies {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: Hobbies', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
