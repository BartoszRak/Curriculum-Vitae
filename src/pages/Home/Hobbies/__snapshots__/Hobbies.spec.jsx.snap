// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Hobbies should render as expected 1`] = `
<section>
  <ContainerMock
    classes={
      Object {
        "root": undefined,
      }
    }
    id="hobbies"
  >
    <TypographyMock
      color="secondary"
      variant="h1"
    >
      <FormattedMessage
        id="pages.home.hobbies.title"
        values={Object {}}
      />
      <span
        aria-hidden={true}
      >
         ðŸ™„
      </span>
    </TypographyMock>
    <div>
      <article>
        <TypographyMock
          variant="h2"
        >
          <FormattedMessage
            id="pages.home.hobbies.tiles.insects.title"
            values={Object {}}
          />
        </TypographyMock>
        <TypographyMock
          variant="body1"
        >
          <FormattedMessage
            id="pages.home.hobbies.tiles.insects.description"
            values={Object {}}
          />
        </TypographyMock>
        <img
          alt="Assasing bug"
          src="assasin-bug.jpg"
        />
      </article>
      <article>
        <TypographyMock
          variant="h2"
        >
          <FormattedMessage
            id="pages.home.hobbies.tiles.programming.title"
            values={Object {}}
          />
        </TypographyMock>
        <TypographyMock
          variant="body1"
        >
          <FormattedMessage
            id="pages.home.hobbies.tiles.programming.description"
            values={Object {}}
          />
        </TypographyMock>
        <img
          alt="Coding"
          src="coding.jpg"
        />
      </article>
    </div>
  </ContainerMock>
</section>
`;
